### ISSUE TITLE
First of all thank you for the great library. I am having a problem generating image file

### ISSUE THREAD

**Issue by hiorti** [kreativekorp/barcode#27]
First and foremost, it's a great library - Thank you.

I am having a bit problem. 

When I call the barcode.php file using GET or POST: i.e. https://..../barcode.php?f=png&s=upc-a&d=213...410945 I am getting the png image. All looks good. 

However, when I try to create an image (png, gif or jpg) using:
$image = $generator->render_image($symbology, $data, $options);
imagepng($image);
imagedestroy($image);

I get this: "�PNG  IHDR�d��� pHYs���+-IDATx����n�0EQ\���}��8����ZR�0��0)��:A����}DAAQDAAQ�FQJ��پ�u��g�[n{v��Ό{�xG�_[w�=[�5�Q�"

I did some reading and it looks like this issue could possibly be due to [UTF8 BOM] output. And I am not sure how to get No-BOM UTF8 output.

I would much appreciate any help.

Thank you in advance.

Kind regards.

**Comment by RebeccaRGB** [comment]
You also have to set the mime type in the header:

header('Content-Type: image/png');

**Comment by hiorti** [comment]
@RebeccaRGB Thanks for the comment. However, I tried that and it didn't help. I tried the mime type in the header inside the php code, as well as inside the header template page where all the page headers are generated.

I am temporarily using the output from direct GET barcode.php results, using this line: 

`echo '<img style="display: inline;-webkit-user-select: none;margin: auto;background-color: hsl(0, 0%, 90%);transition: background-color 300ms;" src="https://.....com/barcode.php?f=png&amp;s=upc-a&amp;d=' . $products_upc . '">';
`

And even though that works fine, I am curious to find out the reason it doesn't work using  

```
$image = $generator->render_image($symbology, $data, $options);
imagepng($image);
imagedestroy($image);

```
I use it inside a php generated page, where I am trying to show the product barcode along with the remaining product details.

Thanks again.
